aspnet/security/anti-request-forgery.md

```diff
@@ -0,0 +1,85 @@
+
+David Paquette gave me permission to copy his blog if we add him to the authors list.
+
+Cross-site request forgery (also known as XSRF or CSRF) is an attack against web-hosted applications whereby a malicious web site can influence the interaction between a client browser and a web site trusted by that browser. These attacks are made possible because web browsers send authentication tokens automatically with every request to a web site.
+
+Here is an example of a CSRF attack:
+
+1. A user logs into `www.example.com`, using forms authentication.
+
+2. The server authenticates the user and issues a  response that includes an authentication cookie.
+
+3. The attacker uses social engineering to trick the user into clicking on a link to a malacious site.
+
+   This malicious site contains the following HTML form:
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+````html
+
+   <h1>You Are a Winner!</h1>
+     <form action="http://example.com/api/account" method="post">
+       <input type="hidden" name="Transaction" value="withdraw" />
+       <input type="hidden" name="Amount" value="1000000" />
+     <input type="submit" value="Click Me"/>
+   </form>
+   ````
+
+Notice that the form action posts to the vulnerable site, not to the malicious site. This is the “cross-site” part of CSRF.
+
+4. The user clicks the submit button. The browser includes the authentication cookie with the request.
+
+5. The request runs on the server with the user’s authentication context, and can do anything that an authenticated user is allowed to do.
+
+Although this example requires the user to click the form button, the malicious page could just as easily run a script that sends an AJAX request. Moreover, using SSL does not prevent a CSRF attack, because the malicious site can send an “[https://](https://)” request.
+
+Typically, CSRF attacks are possible against web sites that use cookies for authentication, because browsers send all relevant cookies to the destination web site. However, CSRF attacks are not limited to exploiting cookies. For example, Basic and Digest authentication are also vulnerable. After a user logs in with Basic or Digest authentication. the browser automatically sends the credentials until the session ends.
+
+  ## IAntiforgery
+
+`IAntiforgery` provides access to the antiforgery system and is exposed in the `Configure` method of the `Startup` class:
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+````c#
+
+   public void Configure(IApplicationBuilder app, IAntiforgery antiforgery)
+   ````
+
+  ## IAntiforgeryAdditionalDataProvider
```

Comment:
This, in concert with the change above, could then become "Extending Anti-Forgery"